src/System.Reflection/tests/PropertyInfo/PropertyInfo_GetSetValueTests.cs

```diff
@@ -0,0 +1,374 @@
+        {
+            TypeWithProperties obj = new TypeWithProperties();
+            PropertyInfo pi = GetProperty(typeof(TypeWithProperties), propertyName);
+
+            Assert.Equal(pi.Name, propertyName);
+
+            Assert.Throws<ArgumentException>(() =>
+            {
+                pi.SetValue(obj, setValue, index);
+            });
+        }
+        
+        //Try to set instance Property with null object 
+        [Theory]
+        [InlineData("HasSetterProp", null, null)]
+        [InlineData("Item", "pw", new Object[] { 99, 2, "SOME string" })]
+        public static void TestSetNull_ThrowsException(string propertyName, Object setValue, Object[] index)
+        {
+            PropertyInfo pi = GetProperty(typeof(TypeWithProperties), propertyName);
+
+            Assert.Equal(pi.Name, propertyName);
+
+            // Generic Exception is thrown in WP8 instead of TargetException
+            Assert.ThrowsAny<Exception>(() => pi.SetValue(null, null, null));
+        }
+
+        //Try to set instance Property with wrong type
+        [Fact]
+        public static void TestSet_ThrowsException()
+        {
+            string propertyName = "HasSetterProp";
+            TypeWithProperties obj = new TypeWithProperties();
+            PropertyInfo pi = GetProperty(typeof(TypeWithProperties), propertyName);
+
+            Assert.Equal(pi.Name, propertyName);
+
+            // Generic Exception is thrown in WP8 instead of TargetException
+            Assert.ThrowsAny<Exception>(() => pi.SetValue(obj, "foo", null));
+        }
+        
+
+        //Try to set indexer Property , Incorrect number of parameters
+        [Fact]
+        public static void TestSet_ThrowsTargetParameterCountException()
+        {
+            string propertyName = "Item";
+            TypeWithProperties obj = new TypeWithProperties();
+            PropertyInfo pi = GetProperty(typeof(TypeWithProperties), propertyName);
+
+
+            Assert.Equal(pi.Name, propertyName);
+
+            Assert.Throws<TargetParameterCountException>(() =>
+            {
+                pi.SetValue(obj, "pw", new Object[] { 99, 2, new string[] { "SOME string" } });  // Incorrect number of parameters
+            });
+        }
+       
+
+
+        //Gets PropertyInfo object from a Type
+        public static PropertyInfo GetProperty(Type t, string property)
+        {
+            TypeInfo ti = t.GetTypeInfo();
+            IEnumerator<PropertyInfo> allproperties = ti.DeclaredProperties.GetEnumerator();
+            PropertyInfo pi = null;
+
+            while (allproperties.MoveNext())
+            {
+                if (allproperties.Current.Name.Equals(property))
+                {
+                    //found property
+                    pi = allproperties.Current;
+                    break;
+                }
+            }
+            return pi;
+        }
+    }
+
+
+    //Reflection Metadata  
+
+    public class MyCoVariantTest
```

Comment:
Nit: I'm not sure this is the best name for this type.  How about something like ```ClassWithObjectArrayProperties```?